### PR TITLE
feat: unification of keypairs

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/2889

Unification of keypairs

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: src/tests/utils.ts [hunk 1]
```diff
@@ -42,34 +36,24 @@ export async function fundAccount(address: KiltKeyringPair['address'], amount: B
...
     authentication,
-    assertion,
+    assertionMethod,
     keyAgreement,
   }
 }
 
 export async function generateDid(account: KiltKeyringPair, mnemonic: string): Promise<DidDocument> {
-  const { authentication, assertion, keyAgreement } = await keypairs(account, mnemonic)
+  ConfigService.get('api')
```
Comment (rflechtner) [inline src/tests/utils.ts:54]: what is this about? can you add a comment explaining why we have this line here?

File: src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts [hunk 2]
```diff
@@ -40,13 +40,10 @@ describe('Well Known Did Configuration integration test', () => {
   beforeAll(async () => {
     const address = await startContainer()
     await connect(address)
-  }, 20_000)
-
-  beforeAll(async () => {
```
Comment (rflechtner) [inline src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts:45]: the split in the two beforeAlls meant that we had 20 seconds allowed to starting the container, and then 30 seconds allotted to the setup. now we have 30 seconds for everything.

**Review by rflechtner — APPROVED** [review]
I like it! Just some minor issues

File: src/tests/utils.ts (hunk 1 above)
Comment (Dudleyneedham) [reply src/tests/utils.ts:54]: I don't think we need this. I can remove it

Comment (Dudleyneedham) [reply src/tests/utils.ts:54]:
```suggestion
```

File: src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts (hunk 2 above)
Comment (Dudleyneedham) [reply src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts:45]: Do you think I should revert this change?

Comment (rflechtner) [reply src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts:45]: I think so, it had its reason

File: src/tests/utils.ts [hunk 3]
```diff
@@ -42,34 +34,22 @@ export async function fundAccount(address: KiltKeyringPair['address'], amount: B
...
-    const { secretKey } = keyFromPath(secretKeyPair, path, 'ed25519')
-    return {
-      ...Utils.Crypto.naclBoxPairFromSecret(blake2AsU8a(secretKey)),
-      type: 'x25519',
-    }
-  })()
+export async function keypairs(mnemonic: string) {
+  const authentication = Utils.Crypto.makeKeypairFromUri(mnemonic, 'ed25519')
+
+  const assertionMethod = Utils.Crypto.makeKeypairFromUri(mnemonic, 'ed25519')
```
Comment (arty-name) [inline src/tests/utils.ts:40]: That’s the default
```suggestion
  const authentication = Utils.Crypto.makeKeypairFromUri(mnemonic)

  const assertionMethod = Utils.Crypto.makeKeypairFromUri(mnemonic)
```

**Pull request merged**